interview-questions.md

```diff
@@ -1,58 +1,49 @@
 # ASSESSMENT 4: Interview Practice Questions
+
 Answer the following questions.
 
 First, without external resources. Challenge yourself to answer from memory.
 
-Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.  
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
 1. What are props in React?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: In React, props stands for properties. Props are used as variables to pass down information from a parent class to a child class via component calls (usually from the App.js file to a component). They are called inside of the component using the syntax this.props.variablename.
 
+Researched answer: Props is an object which contains all of the props passed down to the child component. Props may not only be passed from App.js to a child component, but also between components themselves. Props are read only and can't be directly modified inside of the child component.
 
 2. What is a DOM event?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: DOM events are the bridge between the virtual DOM and the real DOM. They are signals that the user has interacted with the website. They are triggered by event listeners, such as .onClick, .onChange, etc. Once triggered, the event listeners will execute code as specified by the developer.
 
+Researched answer: DOM events are signals that something has occured, or is occuring, typically as triggered by the user. They can include mouse events, keyboard events, HTML frame/object events, HTML form events, user interface events, mutation events (pertaining to changes in the strucutre of the document), and progress events. They can also include touch events for touch-enabled devices.
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Object-oriented programming (OOP) relies on using classes and objects to manipulate data for a desired outcome. This is in contast to functional programming, which relies on logic and functions. Functional programming is considered to be less intuitive than OOP, because objects are data structures which were created to mimic how people see real life objects. In this sense, classes are the idea of a thing, whereas objects are the direct implementation of said thing.
```

Comment:
Nicely done. Both these answers are great. Having a working knowledge of the four pillars of OOP is wonderful.